docs/resources/terraform_provider.md

```diff
@@ -0,0 +1,32 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "tharsis_terraform_provider Resource - terraform-provider-tharsis"
+subcategory: ""
+description: |-
+  Defines and manages a Terraform provider.
+---
+
+# tharsis_terraform_provider (Resource)
+
+Defines and manages a Terraform provider.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `group_path` (String) The path of the group where this Terraform provider resides.
+- `name` (String) The name of the Terraform provider.
+- `private` (Boolean) Whether this Terraform provider is hidden from other top-level groups.
```

Comment:
The `private` and `repository_url` attributes can be optional similar to the module resource.